src/server/frontend_wayland/output_manager.cpp

```diff
@@ -365,6 +347,55 @@ auto mf::OutputManager::from_output_transform(int32_t transform) -> std::tuple<M
     return std::make_tuple(orientation, mirror_mode);
 }
 
+auto mir::frontend::OutputManager::to_output_transform(MirOrientation orientation, MirMirrorMode mirror_mode) -> int32_t
+{
+    switch (orientation)
+    {
+        case mir_orientation_normal:
+        {
+            switch (mirror_mode)
+            {
+                case mir_mirror_mode_horizontal:
+                    return mw::Output::Transform::flipped;
+                default:
+                    return mw::Output::Transform::normal;
+            }
+        }
+        case mir_orientation_right:
+        {
+            switch (mirror_mode)
+            {
+                case mir_mirror_mode_horizontal:
+                    return mw::Output::Transform::flipped_90;
+                default:
+                    return mw::Output::Transform::_90;
+            }
+        }
+        case mir_orientation_inverted:
+        {
+            switch (mirror_mode)
+            {
+                case mir_mirror_mode_horizontal:
+                    return mw::Output::Transform::flipped_180;
+                default:
+                    return mw::Output::Transform::_180;
+            }
+        }
+        case mir_orientation_left:
+        {
+            switch (mirror_mode)
+            {
+                case mir_mirror_mode_horizontal:
+                    return mw::Output::Transform::flipped_270;
+                default:
+                    return mw::Output::Transform::_270;
+            }
+        }
+    }
+
+    return mw::Output::Transform::normal;
```

Comment:
[Non blocking] This looks like it would be easier to read as a lookup table:

```c++
    return lookup[orientation][mirror_mode];
```

(Yeah, I know orientation isn't [0, 4), which slightly complicates the lookup table)